src/main/java/com/epam/ta/reportportal/dao/TestItemRepositoryCustomImpl.java

```diff
@@ -107,7 +96,32 @@ public void setDsl(DSLContext dsl) {
 
 		fetchRetries(items);
 
-		return PageableExecutionUtils.getPage(items,
+		return PageableExecutionUtils.getPage(items, testItemPageable, () -> dsl.fetchCount(QueryBuilder.newBuilder(testItemFilter)
+				.addJointToStart(launchesTable,
+						JoinType.JOIN,
+						TEST_ITEM.LAUNCH_ID.eq(fieldName(launchesTable.getName(), ID).cast(Long.class))
+				)
+				.build()));
+	}
+
+	@Override
+	public Page<Long> findIdsByFilter(Queryable launchFilter, Queryable testItemFilter, Pageable launchPageable,
+			Pageable testItemPageable) {
+
+		Table<? extends Record> launchesTable = getLaunchesAsTable(launchFilter, launchPageable);
+
+		List<Long> testItemsIds = TEST_ITEM_IDS_FETCHER.apply(dsl.fetch(QueryBuilder.newBuilder(testItemFilter)
+				.with(testItemPageable)
+				.addJointToStart(launchesTable,
+						JoinType.JOIN,
+						TEST_ITEM.LAUNCH_ID.eq(fieldName(launchesTable.getName(), ID).cast(Long.class))
+				)
+				.wrap()
```

Comment:
You can exclude wrapping from this query, because it's required only if we fetch the full entity. In the inner query only ids are selected and that fully suits our logic. So you can simplify it it like:
dsl.select(fieldName(ID)).from(<-- built query without wrapping -->).fetchInto(Long.class)

Reply:
Thank you, i will try to simplify the code)